sdk/identity/src/token_credentials/client_secret_credentials.rs

```diff
@@ -4,9 +4,55 @@ use chrono::Utc;
 use oauth2::{
     basic::BasicClient, reqwest::async_http_client, AccessToken, AuthType, AuthUrl, Scope, TokenUrl,
 };
-use std::{str, time::Duration};
+use std::{borrow::Cow, str, time::Duration};
 use url::Url;
 
+/// Provides options to configure how the Identity library makes authentication
+/// requests to Azure Active Directory.
+#[derive(Clone, Debug, PartialEq)]
+pub struct TokenCredentialOptions {
+    /// The authority host to use for authentication requests.  The default is
+    /// "https://login.microsoftonline.com".
+    pub authority_host: Option<Cow<'static, str>>,
+}
+
+const DEFAULT_TOKEN_CREDENTIAL_OPTIONS: TokenCredentialOptions = TokenCredentialOptions {
+    authority_host: Some(Cow::Borrowed(authority_hosts::AZURE_PUBLIC_CLOUD)),
+};
+
+impl Default for TokenCredentialOptions {
+    fn default() -> Self {
+        DEFAULT_TOKEN_CREDENTIAL_OPTIONS
+    }
+}
+
+impl TokenCredentialOptions {
+    pub fn authority_host(&self) -> &str {
```

Comment:
Maybe this can be simplified removing `Cow` and `Option`.

Something like:

```rust
 #[derive(Clone, Debug, PartialEq)]
  pub struct TokenCredentialOptions {
      /// The authority host to use for authentication requests.  The default is
      /// "https://login.microsoftonline.com".
      pub authority_host: &'static str,
  }
  
  impl Default for TokenCredentialOptions {
      fn default() -> Self {
          Self {
              authority_host: authority_hosts::AZURE_PUBLIC_CLOUD,
          }
      }   
  }
  
  impl TokenCredentialOptions {
      pub fn authority_host(&self) -> &str {
          self.authority_host
      }   
  }
```

What do you think?

Reply:
I think each option in options is supposed to be optional.

Reply:
That is consistent with the TypeScript TDK. In TypeScript, the `?` marks it optional.
- https://github.com/Azure/azure-sdk-for-js/blob/7e9fcae3430a679cf8f0cfdcc08fe43d7576928b/sdk/identity/identity/src/client/identityClient.ts#L232-L238
- https://github.com/Azure/azure-sdk-for-js/blob/7e9fcae3430a679cf8f0cfdcc08fe43d7576928b/sdk/identity/identity/src/client/identityClient.ts#L232-L238

Reply:
It's a bit pedantic, I know, but I strongly believe in the expressivity of types. 

In this case the authority is not optional, it is a mandatory field: you can't obtain a credential without it.  Your code conveys this message with the getter:

```rust
pub fn authority_host(&self) -> &str {
        match &self.authority_host {
            Some(authority_host) => authority_host,
            None => authority_hosts::AZURE_PUBLIC_CLOUD,
        }
}
```

So why have an `Option` in the struct? If we make it private it won't be seen anyway.

Reply:
It is no longer an Option as requested.